src/datasets/features/features.py

```diff
@@ -1698,14 +1701,14 @@ def unsimplify(feature: dict) -> dict:
             #
             # class_label:              ->              class_label:
             #   names:                  ->                names:
-            #     0: negative              ->               - negative
-            #     1: positive              ->               - positive
+            #     '0': negative              ->               - negative
+            #     '1': positive              ->               - positive
             #
             if isinstance(feature.get("class_label"), dict) and isinstance(feature["class_label"].get("names"), dict):
                 label_ids = sorted(feature["class_label"]["names"])
-                if label_ids and label_ids != list(range(label_ids[-1] + 1)):
+                if label_ids and [int(label_id) for label_id in label_ids] != list(range(int(label_ids[-1]) + 1)):
                     raise ValueError(
-                        f"ClassLabel expected a value for all label ids [0:{label_ids[-1] + 1}] but some ids are missing."
+                        f"ClassLabel expected a value for all label ids [0:{int(label_ids[-1]) + 1}] but some ids are missing."
```

Comment:
```suggestion
                        f"ClassLabel expected a value for all label ids [0:{label_ids[-1] + 1}] but some ids are missing."
```

Reply:
This is also a consequence of the suggested change above.